lib/ui/text.dart

```diff
@@ -1406,6 +1406,90 @@ class TextPosition {
   }
 }
 
+/// A range of characters in a string of text.
+class TextRange {
+  /// Creates a text range.
+  ///
+  /// The [start] and [end] arguments must not be null. Both the [start] and
+  /// [end] must either be greater than or equal to zero or both exactly -1.
+  ///
+  /// Instead of creating an empty text range, consider using the [empty]
+  /// constant.
```

Comment:
Should clarify if a range includes `end` or not

Reply:
Good call.  Will add.